presto-array/src/test/java/com/facebook/presto/array/TestSliceBigArray.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.array;
+
+import io.airlift.slice.Slice;
+import org.openjdk.jol.info.ClassLayout;
+import org.testng.annotations.Test;
+
+import static io.airlift.slice.SizeOf.sizeOf;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static org.testng.Assert.assertEquals;
+
+public class TestSliceBigArray
+{
+    private static final long SLICE_BIG_ARRAY_INSTANCE_SIZE = ClassLayout.parseClass(SliceBigArray.class).instanceSize();
+    private static final long SLICE_INSTANCE_SIZE = ClassLayout.parseClass(Slice.class).instanceSize();
+
+    @Test
```

Comment:
this method is huge and very difficult to follow. [Here](https://gist.githubusercontent.com/nezihyigitbasi/544992d84b328209ae9eb747bf02f648/raw/3561125e4885eef44afccf09889b6c6149a7f2b6/slice-test.java) is how I would rewrite the whole thing.